exec/call_test.go

```diff
@@ -153,3 +190,50 @@ func TestHostSymbolCall(t *testing.T) {
 		t.Fatalf("Did not get the right value. Got %d, wanted %d", rtrns, 3)
 	}
 }
+
+func TestGoFunctionCallChecksForFirstArgument(t *testing.T) {
+	m, err := wasm.ReadModule(bytes.NewReader(moduleCallHost), invalidImporter)
+	if err != nil {
+		t.Fatalf("Could not read module: %v", err)
+	}
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("This code should have panicked.")
+		} else {
+			if r != "exec: the first argument of a host function was int32, expected ptr" {
+				t.Errorf("This should have panicked because of %v", err)
```

Comment:
not sure what `err` will be here.

Reply:
it should be `nil`. That's an interesting mistake, `err` and `r` got mixed up.

Reply:
`err` would be nil, and indeed it's not what we should print